### PR TITLE
1901 MPI_Wtime called many times per handler invocation

### LINKED ISSUE #1901: MPI_Wtime called many times per handler invocation
**Describe the bug**

**To Reproduce**
1. Example/test/snippet of code that fails
2. Compiler, platform, libraries
3. Run command: number of processors, threading options, etc.

**Expected behavior**

**Screenshots**

**Platform (please complete the following information):**
 - Supercomputer or desktop [e.g., Trinity, Sierra)
 - Compiler environment [e.g., clang-3.9, gcc8]
 - Libraries [e.g., mpich-2, openmpi)
 - Commit ID or release of VT [e.g., vt-1.0.0]

**Additional context**

### PR DESCRIPTION
**Pull request by stmcgovern**
Closes #1901. Pass the current time from the scheduler to the progress functions.

### CODE REVIEW

File: src/vt/runtime/component/component.h [hunk 1]
```diff
@@ -186,7 +188,7 @@ struct PollableComponent : Component<T> {
    *
    * \return number of units processed---zero
    */
-  virtual int progress() override {
+  virtual int progress(TimeType current_time) override {
```
Comment (PhilMiller) [inline src/vt/runtime/component/component.h:191]: (not for this PR)

I just noticed - could this runtime abort be made to fail at compile time with `virtual int progress(TimeType current_time) override = 0;`?

File: src/vt/timetrigger/time_trigger_manager.cc [hunk 2]
```diff
@@ -87,8 +87,8 @@ void TimeTriggerManager::triggerReady(TimeType cur_time) {
   }
 }
 
-int TimeTriggerManager::progress() {
-  auto const cur_time = timing::getCurrentTime();
+int TimeTriggerManager::progress(TimeType current_time) {
+  auto const cur_time = current_time;
   triggerReady(cur_time);
```
Comment (PhilMiller) [inline src/vt/timetrigger/time_trigger_manager.cc:92]: Could we just drop the local variable, and pass the parameter directly?

**Comment by PhilMiller** [comment]
Looks good to me other than build failure. We're going to resolve that this afternoon.

File: src/vt/timetrigger/time_trigger_manager.cc (hunk 2 above)
Comment (stmcgovern) [reply src/vt/timetrigger/time_trigger_manager.cc]: done

**Comment by PhilMiller** [comment]
I think I'd vote for merging this sooner rather than later, and opening separate issues for the follow-ups we identified in trace and LB

**Comment by stmcgovern** [comment]
OK, I will open an issue for the follow-ons.

**Review by PhilMiller — APPROVED** [review]

**Comment by PhilMiller** [comment]
Wait for CI to complete so that ccache gets saved, then rebase on `develop` and push that

**Pull request merged**